### PR TITLE
Validator + Fix broken tests

### PR DESCRIPTION
**Pull request by fpenim**
This PR adds BioStudies validation @davidrichardson and fixes the broken tests @fpenim .

### CODE REVIEW

**Comment by davidrichardson** [comment]
👍

**Comment by davidrichardson** [comment]
We should ask @ke4 to review, since @fpenim and I worked on the code

File: src/main/java/uk/ac/ebi/subs/biostudies/validation/ValidationMessaging.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    public static final String BIOSTUDIES_PROJECT_VALIDATION = "biostudies-project-validation";
+    public static final String EVENT_VALIDATION_SUCCESS = "validation.success";
+    public static final String EVENT_VALIDATION_ERROR = "validation.error";
+
+    static final String EVENT_BIOSTUDIES_PROJECT_VALIDATION = "biostudies.project.validation";
+
+
+    /**
+     * RabbitMQ related messaging configuration for the Biosamples queue(s) and binding(s).
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/biostudies/validation/ValidationMessaging.java]: Copy and Paste pattern :)
Please, correct it in the whole class.

File: src/test/java/uk/ac/ebi/subs/biostudies/agent/BioStudiesProcessorTest.java [hunk 2]
```diff
@@ -26,21 +24,20 @@
 
 import static org.mockito.Mockito.when;
 
-@RunWith(SpringRunner.class)
-
+@RunWith(SpringJUnit4ClassRunner.class)
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/biostudies/agent/BioStudiesProcessorTest.java]: `SpringRunner` is just an alias of `SpringJUnit4ClassRunner` and it is shorter and sounds cooler ;)
https://docs.spring.io/spring/docs/4.3.0.RC2_to_4.3.0.RELEASE/Spring%20Framework%204.3.0.RELEASE/org/springframework/test/context/junit4/SpringRunner.html

File: src/test/java/uk/ac/ebi/subs/biostudies/client/BioStudiesClientTest.java [hunk 3]
```diff
@@ -62,19 +68,19 @@ public void loginFailure() {
         BioStudiesSession session = client.initialiseSession();
 
         //don't expect to get here
-        Assert.assertNotNull(session);
+        assertNotNull(session);
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/biostudies/client/BioStudiesClientTest.java]: I think we don't need this line. We are expecting an exception in this test. If we won't get it, than the test will fail anyway.

File: src/test/java/uk/ac/ebi/subs/biostudies/client/BioStudiesClientTest.java [hunk 4]
```diff
@@ -88,10 +94,11 @@ public void updateGood() {
         bioStudiesSubmission.setAccno("SUBSPRJ1");
 
         SubmissionReport response = session.update(bioStudiesSubmission);
+        System.out.println(response);
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/biostudies/client/BioStudiesClientTest.java]: Do we need this line?

File: src/test/java/uk/ac/ebi/subs/biostudies/validator/ProjectValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    static Project project() {
+        Project p = new Project();
+        p.setTeam(Team.build("test"));
+        p.setAlias("alias");
+
+        return p;
+    }
+
+    static ValidationMessageEnvelope<Project> generateValidationMessageEnvelope(Project project) {
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/biostudies/validator/ProjectValidatorTest.java]: Are we using this method somewhere? I could not find it.

**Review by ke4 — CHANGES_REQUESTED** [review]
Just tiny things, but it looks OK.
My general question: are we keeping the agent and the validator together in one repo? This was not the case for biosamples.

File: src/test/java/uk/ac/ebi/subs/biostudies/agent/BioStudiesProcessorTest.java (hunk 2 above)
Comment (fpenim) [reply src/test/java/uk/ac/ebi/subs/biostudies/agent/BioStudiesProcessorTest.java]: I didn't know that 👍

File: src/test/java/uk/ac/ebi/subs/biostudies/validator/ProjectValidatorTest.java (hunk 5 above)
Comment (fpenim) [reply src/test/java/uk/ac/ebi/subs/biostudies/validator/ProjectValidatorTest.java]: Indeed we are not

**Review by ke4 — APPROVED** [review]
Looks fine 👍

**Pull request merged**